code/modules/mob/mob_grab.dm

```diff
@@ -116,87 +215,152 @@
+			switch(assailant.a_intent)
+				if("help")
+					/*if(force_down)
+						assailant << "<span class='warning'>You no longer pin [affecting] to the ground.</span>"
+						force_down = 0
+						return*///This is a very basic demonstration of a new feature based on attacking someone with the grab, based on intent.
+								//This specific example would allow you to stop pinning people to the floor without moving away from them.
+					return
+
+				if("grab")
+					return
+
+				if("harm") //This checks that the user is on harm intent.
+					/*if(last_hit_zone == "eyes")
+						if(state < GRAB_NECK)
+							assailant << "<span class='warning'>You require a better grab to do this.</span>"
+							return
+						if((affected.head && affected.head.flags & HEADCOVERSEYES) || \
+							(affected.wear_mask && affected.wear_mask.flags & MASKCOVERSEYES) || \
+							(affected.glasses && affected.glasses.flags & GLASSESCOVERSEYES))
+							assailant << "<span class='danger'>You're going to need to remove the eye covering first.</span>"
+							return
+						if(!affected.internal_organs_by_name["eyes"])
+							assailant << "<span class='danger'>You cannot locate any eyes on [affecting]!</span>"
+							return
+						assailant.visible_message("<span class='danger'>[assailant] presses \his fingers into [affecting]'s eyes!</span>")
+						affecting << "<span class='danger'>You feel immense pain as digits are being pressed into your eyes!</span>"
+						assailant.attack_log += text("\[[time_stamp()]\] <font color='red'>Pressed fingers into the eyes of [affecting.name] ([affecting.ckey])</font>")
+						affecting.attack_log += text("\[[time_stamp()]\] <font color='orange'>Had fingers pressed into their eyes by [assailant.name] ([assailant.ckey])</font>")
+						msg_admin_attack("[key_name(assailant)] has pressed his fingers into [key_name(affecting)]'s eyes.")
+						var/obj/item/organ/eyes/eyes = affected.internal_organs_by_name["eyes"]
+						eyes.damage += rand(3,4)
+						if (eyes.damage >= eyes.min_broken_damage)
+							if(M.stat != 2)
+								M << "\red You go blind!"*///This is a demonstration of adding a new damaging type based on intent as well as hitzone.
+															//This specific example would allow you to squish people's eyes with a GRAB_NECK.
+					if(last_hit_zone == "head") //This checks the hitzone the user has selected. In this specific case, they have the head selected.
+						if(affecting.lying)
+							return
+						assailant.visible_message("<span class='danger'>[assailant] thrusts \his head into [affecting]'s skull!</span>") //A visible message for what is going on.
+						var/damage = 5
+						var/obj/item/clothing/hat = attacker.head
+						if(istype(hat))
+							damage += hat.force * 3
+						affecting.apply_damage(damage*rand(90, 110)/100, BRUTE, "head", affected.run_armor_check(affecting, "melee"))
+						assailant.apply_damage(3*rand(90, 110)/100, BRUTE, "head", attacker.run_armor_check("head", "melee"))

```

Comment:
Personally, I'd just remove the self damage now; 5 damage is low enough that there doesn't really need to be a downsides to headbutting, especially when the average unarmed melee damage is 4.5